### PR TITLE
AUTH-509: remove unauthenticated group from any cluster role binding

### PR DESCRIPTION
**Pull request by ibihim**
## What

Remove UnauthenticatedGroup from:

- SelfAccessReview
- OAuthTokenDelete
- WebHooksRole
- ScopeImpersonation

## Why

An unauthenticated User should have no permissions.

### CODE REVIEW

**Comment by deads2k** [comment]
add localresourceaccessreviews to https://github.com/openshift/openshift-apiserver/blob/a5318cdb1de427577f218e1b6237fd8c26003103/pkg/bootstrappolicy/policy.go#L403

File: pkg/bootstrappolicy/policy.go [hunk 1]
```diff
@@ -425,6 +425,9 @@ func GetOpenshiftBootstrapClusterRoles() []rbacv1.ClusterRole {
 
 				rbacv1helpers.NewRule(read...).Groups(templateGroup, legacyTemplateGroup).Resources("templates", "templateconfigs", "processedtemplates", "templateinstances").RuleOrDie(),
 
+				// after dropping unauthenticated groups
+				rbacv1helpers.NewRule("create").Groups(authzGroup, legacyAuthzGroup).Resources("localresourceaccessreviews").RuleOrDie(),
```
Comment (deads2k) [inline pkg/bootstrappolicy/policy.go:429]: I got the wrong name here.  We'd like to keep the "can I do this in a namespace" with namespace reader.  And the `selfsubjectrulesreviews` (what can I do in this namespace)

File: pkg/bootstrappolicy/policy.go [hunk 2]
```diff
@@ -425,6 +425,15 @@ func GetOpenshiftBootstrapClusterRoles() []rbacv1.ClusterRole {
 
 				rbacv1helpers.NewRule(read...).Groups(templateGroup, legacyTemplateGroup).Resources("templates", "templateconfigs", "processedtemplates", "templateinstances").RuleOrDie(),
 
+				// after dropping unauthenticated groups
+				rbacv1helpers.NewRule("create").Groups(authzGroup, legacyAuthzGroup).Resources(
+					"selfsubjectaccessreviews",
+					"subjectaccessreviews",
+
+					"subjectrulesreviews",
```
Comment (deads2k) [inline pkg/bootstrappolicy/policy.go:433]: remove

File: pkg/bootstrappolicy/policy.go [hunk 3]
```diff
@@ -425,6 +425,15 @@ func GetOpenshiftBootstrapClusterRoles() []rbacv1.ClusterRole {
...
+					"selfsubjectrulesreviews",
```
[the remark this answers is not in the document]
Comment (ibihim) [reply pkg/bootstrappolicy/policy.go:435]: There is a German proverb: doppelt hält besser.

[the remark this answers is not in the document]
Comment (ibihim) [reply pkg/bootstrappolicy/policy.go:432]: The first is selfsubject**access**reviews and the other is self**rules**reviews.

**Pull request merged**